network-mux/src/Network/Mux/DeltaQTraceTransformer.hs

```diff
@@ -1,21 +1,28 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
 module Network.Mux.DeltaQTraceTransformer
 where
 
-import Data.IntMap (IntMap)
-import qualified Data.IntMap as IM
-import Data.Maybe
+import           Data.IntMap.Strict (IntMap)
+import qualified Data.IntMap.Strict as IM
+import           Data.Maybe
 
-import Control.Monad.Class.MonadTime
-import Network.Mux.Types
+import           Control.Monad.Class.MonadTime
+import           Network.Mux.Types
 
 -- Map time intervals to real numbers, for the arithmetic.
 newtype SISec  = S  Float -- this is all the precision we need,
+  deriving (Eq, Ord, Num)
 newtype SISec2 = S2 Float -- are there performance reasons to use Double?
+  deriving (Eq, Ord, Num)
 
+squareSISec :: SISec -> SISec2
+squareSISec (S x) = S2 $ x * x
+
+-- the observation step
 step :: StatsA -> RemoteClockModel -> Time -> Int -> (Maybe Sample, StatsA)
 step s remoteTS localTS obsSize
  | isNothing (referenceTimePoint s)
-   = step (s { referenceTimePoint = Just (remoteTS, localTS)
+   = step (s { referenceTimePoint = Just $! (remoteTS, localTS)
```

Comment:
This `$!` does nothing. Quiz: why? :smile:

Reply:
because the pair is not strict?

Reply:
Because $! simply evaluates to WHNF, and `(,)` is already in WHNF.